portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +27,54 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private List<String> comments;
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Chance!");
-  }
+    @Override
+    public void init() {
+        comments = new ArrayList<>();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+
+        response.setContentType("application/json");
+        response.getWriter().println(json);
+    }
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String text = getParameter(request, "text-input", "");
+        boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+        boolean sort = Boolean.parseBoolean(getParameter(request, "sort", "false"));
+
+        // Convert the text to upper case.
+        if (upperCase) {
+            text = text.toUpperCase();
+        }
+
+        // Break the text into individual words.
+        String[] words = text.split("\\s*,\\s*");
+
+        // Sort the words.
+        if (sort) {
+            Arrays.sort(words);
+        }
+        comments.add(text);
+        comments.add(Arrays.toString(words));
+
+        // Respond with the result.
+        response.setContentType("text/html;");
+        response.getWriter().println(comments);
+        // response.sendRedirect("/index.html");
```

Comment:
nit: remove dead code

Reply:
oops, not dead code, should not have been commented out. Will fix.